src/Compilers/Core/Portable/CommandLine/CommonCompiler.CompilerEmitStreamProvider.cs

```diff
@@ -37,18 +37,72 @@ public void Dispose()
             public override Stream CreateStream(DiagnosticBag diagnostics)
             {
                 Debug.Assert(_streamToDispose == null);
+                bool renameAttempted = false;
 
-                try
+                while (true)
```

Comment:
This is a loop that can essentially execute twice.  We shouldn't be using a loop here.  Especially `while (true)`.  Would strongyl prefer a method with a parameter that is switched when we hit the case so that the two stage nature of the delete is explicit